internal/controller/clusterinstance_controller_test.go

```diff
@@ -468,20 +496,594 @@ var _ = Describe("handleFinalizer", func() {
 
 })
 
+type pruneTestStruct struct {
+	Object      client.Object
+	Key         types.NamespacedName
+	ShouldPrune bool
+}
+
+var _ = DescribeTable("pruneManifests",
```

Comment:
I think this test might be a lot easier to understand if it were three different examples with a `BeforeEach`.

I don't think we're gaining much from the table test here except not being able to see the resources we're using and what we're doing with them at the same time.

Reply:
The primary reason for using table test is to avoid repeating functionality that would have to be added for each test case for both the test setup and then the test evaluation. The "entries" describe the test being performed, whereas the `pruneTestStruct` defines the unique data values / arguments for each test case.

Reply:
Yeah, I know why you would do this, but I tend to shy away from table tests when the entries become this large or when the test function has too many conditions.

To me it feels more clear to split the different behavior.

Maybe it would be enough to use a `BeforeEach` in the `DescribeTable` to separate common setup from the important bits of the test function.

I'll leave it up to you either way. This is very much a personal preference.

Reply:
Refactored the test!